base/task.jl

```diff
@@ -197,6 +215,7 @@ function produce(v)
     t.state == :runnable || throw(AssertionError("producer.consumer.state == :runnable"))
     if empty
         schedule_and_wait(t, v)
+        ct = current_task() # When a task is copied, ct should be updated to new task ID.
```

Comment:
No - this is NOT for copying running task. It is needed for correctly copying suspended but not-finished task. Here is why:

When a task `t` calls the `produce' function, control flow is passed to another task, say we call this task `s`. Notice that before the control flow is passed to `s`, a variable called `ct` which stores a reference to `t` is created on task `t`'s stack. Say now we create a copy of task `t` and call the duplicated task `newt`, the `ct` variable on both `t` and `ct`'s stack will have the same value (task `t`). 

Now if we want to run task `newt`, we need to call some version of `consume(newt)`, since the `ct` variable on `newt`'s stack points to the original task `t`, the result is we incorrectly pass control flow to `t` instead of `ct.